### ISSUE TITLE
Tax amounts from Adjustments on regular invoice not displayed correctly

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#7983]
## Description

Follow-up issue on point 6), see [testing notes here](https://github.com/openfoodfoundation/openfoodnetwork/pull/7805#issuecomment-886766743).

Relates to #7933.

There are some inconsistencies introduced by PR #7805 in what concerns the displaying of Tax amounts on the regular invoice, for both inclusive and added-tax instances, which can be confusing to users:

(UPDATED to reflect the improvements introduced on #7989)

- `Line items` do not display any value on `Sales Tax`~~/`GST`~~
~~- `Shipping` displays 0.00 despite having an associated tax value~~
- Added-tax instances: an `added adjustment` displays the correct tax value
- Inclusive-tax instances: an `added adjustment` displays the correct tax value but additionally displays the wording `(included)`; this entry is not centered on the column.

## Expected Behavior

**Added-tax instances**

The regular invoice should display the `Sales Tax` correctly for all line items, shipping and added adjustments correctly or (instead) this should be omitted for all of these.

**Inclusive-tax instances**

The regular invoice should display the `GST` correctly for all line items, shipping and added adjustments correctly:
- the wording `(included)` appears next to the GST value; this entry is not centered

## Actual Behaviour

Information is displayed inconsistently on the regular invoice (see pics).

## Steps to Reproduce

1. Place orders where tax is applied for both added- and inclusive-tax instances.
2. Check the regular invoice, for the respective instances, namely:
3. Added-tax -> the `Sales Tax` column; Inclusive-tax -> the `GST` column
4. Notice the inconsistencies (pics below)

## Animated Gif/Screenshot

- Added-tax -> the `Sales Tax` column

![image.png](https://images.zenhubusercontent.com/5d761b049dde650001eaf32f/aba289ce-9f67-4320-93bb-fd852dca34a4)

- Inclusive-tax -> the `GST` column

![image.png](https://images.zenhubusercontent.com/5d761b049dde650001eaf32f/f020a7c4-4b91-429d-aed7-7a34731832c1)

## Workaround

A bit confusing.

## Severity

bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used: v3.8.7
* Browser name and version: Firefox 90
* Operating System and version (desktop or mobile): Ubuntu 20.04

**Comment by tschumilas** [comment]
WOW - chose this as Canada's s3 -   there would be lots of happy campers here (despite the raging forest fires everywhere) if we had taxes displayed for line items and shipping on invoices!!!!!

**Comment by RachL** [comment]
@tschumilas actually this is already in code review. You get to choose another one ;)

**Comment by tschumilas** [comment]
BONUS!!!

**Comment by jibees** [comment]
**1. Check why the shipping tax is always displayed at 0 even though there are shipping taxes**

I couldn't reproduce that particular case.

With a Shipping Tax with amount at `0.2`, I've the following:
<img width="608" alt="Capture d’écran 2022-07-28 à 16 48 41" src="https://user-images.githubusercontent.com/296452/181566486-5784841e-dee2-4fd0-9c86-7ac378cf7aec.png">
<img width="751" alt="Capture d’écran 2022-07-28 à 16 45 40" src="https://user-images.githubusercontent.com/296452/181566701-6193a849-6dac-4f0a-97b5-504172645aeb.png">

And if the Shipping Tax is at a `0` rate, then I got: 
<img width="745" alt="Capture d’écran 2022-07-28 à 16 48 25" src="https://user-images.githubusercontent.com/296452/181567086-5e6439ce-7308-436c-a2c0-a81058890f5b.png">

**Comment by lin-d-hop** [comment]
> Display the tax on line items when taxes are included in prices
Yes please this is the standard way tax is displayed on invoices when included in price.

**Comment by filipefurtad0** [comment]
Uff - this has been a while and I needed another dive in this.

To my understanding, the issue concerns only instances like CA which:

- have tax added to the price (i.e. not included)
- use the standard invoice

On the points you raise:

> 1.  Check why the shipping tax is always displayed at 0 even though there are shipping taxes

This has been solved as pointed out [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/7989#issuecomment-895976569) and scratched out from the description - but the picture I posted before was not updated - apologies for the confusion. @jibees's pic above nicely demonstrates this has been fixed :+1: 

> 2. Display the tax on line items when taxes are included in prices: @lin-d-hop do you confirm this is the way to go when prices are including tax? I guess it will impact UK's invoice template.

Any PR related to this should not introduce any changes to the UK's invoice (included tax + standard invoice), as I believe this is working correctly for the UK, although there are some issues still - like [this one](https://github.com/openfoodfoundation/wishlist/issues/176) - the tax amounts seem to be currently displayed correctly, per line item :+1: 

It is only for CA (added tax + standard invoice) that they are not. Again, this is seen on JB's pic, the missing GST value for the line item Tomatoes (and for the Admin and Handling fee, if applicable).

In summary, commenting on JBs pick:

![image](https://user-images.githubusercontent.com/49817236/181619600-69ed586b-a112-4db5-8bce-037d47eb8686.png)

**Comment by jibees** [comment]
Hey @filipefurtad0 
Thanks a lot for your comment, very useful.

Sorry, but there is another noob question, I just want to be sure I've the right configuration regarding what you meant... Here's my configuration:

1. Hug charge GST
<img width="385" alt="Capture d’écran 2022-07-29 à 10 45 03" src="https://user-images.githubusercontent.com/296452/181721232-a79fd553-001b-4279-99ff-faddff5b4a3b.png">

2. I've two Tax categories, both not included in price
<img width="877" alt="Capture d’écran 2022-07-29 à 10 38 09" src="https://user-images.githubusercontent.com/296452/181721301-59738d12-7451-42be-bc1e-d2d9cf8cbdbf.png">

3. The hub is selling tomatoes, using the "Tax Category"
<img width="1084" alt="Capture d’écran 2022-07-29 à 10 38 24" src="https://user-images.githubusercontent.com/296452/181721387-6477b6df-6544-45fa-bcb8-307d43a8666d.png">

4. The shipping method used in the order use the "Shipping" tax
<img width="728" alt="Capture d’écran 2022-07-29 à 10 39 13" src="https://user-images.githubusercontent.com/296452/181721462-f715d94d-1416-4440-8a35-dc6d72c7957c.png">

5. When I checkout, here the order summary:
<img width="1037" alt="Capture d’écran 2022-07-29 à 10 40 09" src="https://user-images.githubusercontent.com/296452/181721623-355044cf-cdd1-40b4-a0dd-ce5f6517c1e3.png">

6. In the admin (I've soooo many fees, used for another issue, maybe I should delete them all):
<img width="1056" alt="Capture d’écran 2022-07-29 à 10 40 36" src="https://user-images.githubusercontent.com/296452/181721660-6100c9d0-7758-423d-a83f-4d83f774a0db.png">

7. When I print the invoice (something has changed between my last screenshot, I've lost the GST value for the Shipping line....):
<img width="735" alt="Capture d’écran 2022-07-29 à 10 41 17" src="https://user-images.githubusercontent.com/296452/181721801-b10b42eb-7c96-459f-ae93-7ac3a8a41411.png">

What do you think?

**Comment by filipefurtad0** [comment]
Thanks JB,

>  I've lost the GST value for the Shipping line.... 

That's a bit mysterious... Ok, I'll try to reproduce this in staging and then in a spec - ideally? This way we have a specific configuration which we could run and see if it breaks, upon any changes.

I wonder how does the adjustments tab `/admin/orders/<order_number>/adjustments` look like? Investigating.

> ok, seems like we display here the GST value when taxes are included in price

Great, that seems to make sense with the current behavior :+1:

**Comment by filipefurtad0** [comment]
Adding to my previous comment: I can only reproduce [your previous findings]( https://github.com/openfoodfoundation/openfoodnetwork/issues/7983#issuecomment-1198251293) :+1: My setup is in the [attached PDF file](https://github.com/openfoodfoundation/openfoodnetwork/files/9219199/tax_not_included_legacy_invoice.pdf).

Now on to put this on a spec.

**Comment by jibees** [comment]
> Now on to put this on a spec: spec/features/admin/invoice_print_spec.rb seems suitable although we can't really assert on formatting and only on the html content. But I think it should be fine if the values are carefully selected, let's see how it goes.

I agree that analyzing the content should be enough, as we probably can select _easily_ values inside the `<table />`. Would you mind create that spec?
